optimum/amd/brevitas/configuration.py

```diff
@@ -0,0 +1,105 @@
+        scale_precision (`str`, defaults to `"float_scale"`):
+            Precise the constraints on the scale. Can either be `"float_scale"` (arbitrary scales), or `"power_of_two_scale"` (scales constrainted to be a power of 2).
+        weights_quant_granularity (`str`, defaults to `"per_tensor"`):
+            The granularity of the quantization of the weights. This parameter can either be:
+            - `"per_tensor"`: A single scale (and possibly zero-point) is used for one weight matrix.
+            - `"per_channel"`: Each column (outer dimension) of the weight matrix has its own scale (and possibly zero-point).
+            - `"per_group"`: Each column of the weight matrix may have several scales, grouped by `weight_group_size`.
+        weights_group_size (`Optional[int]`, defaults to `None`):
+            Group size to use for the weights in case `weights_quant_granularity="per_group"`. Defaults to `128` in this case, to `None` otherwise.
+        quantize_zero_point (`bool`, defaults to `True`):
+            When set to True, the unquantized value 0.0 is exactly representable as a quantized value: the zero point. When set to False, a quantization range [a, b] is exactly reprensentable (no rounding on a and b), but the unquantized value zero is not exactly representable.
+        activations_param_method (`List[str]`):
+            Strategy to use to estimate the quantization parameters (scale, zero-point) for the activations. Two strategies are available:
+            - `"stats"`: Use min-max to estimate the range to quantize on.
+            - `"mse"`: Use mean-square error between the unquantized activations and quantized activations to estimate the range to quantize on.
+        is_static (`bool`, defaults to `False`):
+            Whether to apply static quantization or dynamic quantization.
+        activations_symmetric (`bool`, defaults to `False`):
+            Whether to use symmetric quantization on the activations.
+        activations_quant_granularity (`str`, defaults to `"per_tensor"`):
+            The granularity of the quantization of the activations. This parameter can either be `"per_tensor"`, `"per_row"` or `"per_group"`. In case static quantization is used (`is_static=True`), only `"per_tensor"` and `"per_group"` may be used.
+        activations_group_size (`int`, defaults to `None`):
+            Group size to use for the activations in case `activations_quant_granularity="per_group"`. Defaults to `64` in this case, to `None` otherwise.
+        activations_equalization (`Optional[str]`, defaults to `"cross_layer"`):
+            Whether to apply apply activation equalization (SmoothQuant). Possible options are:
+            - `None`: No activation equalization.
+            - `"cross_layer"`: Use Brevitas GraphActivationEqualization for activation and weight equalization.
+            - `"layerwise"`: Apply SmoothQuant as described in https://arxiv.org/abs/2211.10438.
```

Comment:
I think neither of these options fuse the activation rescaling in e.g. the preceding layernorm https://github.com/mit-han-lab/smoothquant/blob/40e55f7f2ce8b3068eea5c1356aead9b70405bc0/smoothquant/opt.py#L232-L233, or I did not find the code for that

Reply:
When calling `activation_equalization_mode` with `layerwise=False`, [this class](https://github.com/Xilinx/brevitas/blob/925545b0d2d8ebc58544bbd2c6e3be8bb86a5b65/src/brevitas/graph/equalize.py#L1031) is instantiated, which will attach hooks to compute statistics for activation equalization. 

Then we use these statistics [the same way](https://github.com/Xilinx/brevitas/blob/925545b0d2d8ebc58544bbd2c6e3be8bb86a5b65/src/brevitas/graph/equalize.py#L1114) we would do in weight equalization, i.e. we combine them with the ones coming from the subsequent layer and then we merge the resulting scale factor with the previous/subsequent layers.